javascript/challenges/mutliBracketValidation/multi-bracket-validation.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+function isProperPair(opener, closer) {
+  switch(opener) {
+  case '[' :
+    return closer === ']';
+  case '{' :
+    return closer === '}';
+  case '(' :
+    return closer === ')';
+  }
+  return false;
+}
+
+function multiBracketValidation(str) {
+
+  if(!str) {
+    return false;
+  }
+  const openers = '{(['
+  const closers = '})]'
+  let stack = [];
+
+  for (let i = 0; i< str.length; i++) {
+    let currentChar = str[i];
+    if(openers.includes(currentChar)) {
+      stack.push(currentChar);
+    } else if (closers.includes(currentChar)) {
+      const poppedChar = stack.pop();
+
+      if(!isProperPair(poppedChar, currentChar)) {
+        return false;
+      }
+    }
+  }
+  return stack.length === 0;
+}
+
+module.exports = multiBracketValidation;
+
```

Comment:
Something I would suggest doing is splitting the string that you are taking in, if you find an opening character, push that into the empty array and then do some comparison if you find a closing bracket and so on